scripts/build_test.py

```diff
@@ -647,4 +653,58 @@ def test_generate_delete_tasks_to_remove_deleted_files(self):
         self.assertEqual(
             len(delete_tasks), build.get_file_count(MOCK_TEMPLATES_DEV_DIR))
 
+    def test_compiled_js_dir_validation(self):
+        """Test that build.COMPILED_JS_DIR is validated correctly with
+        outDir in tsconfig.json.
+        """
+        build.test_compiled_js_dir_is_valid()
+
+        out_dir = ''
+        with open('tsconfig.json') as f:
+            config_data = json.load(f)
+            out_dir = os.path.join(config_data['compilerOptions']['outDir'], '')
+        with self.assertRaisesRegexp(
+            Exception,
+            'COMPILED_JS_DIR: %s is does not match the output directory '
+            'in tsconfig.json: %s' % (
+                MOCK_COMPILED_JS_DIR, out_dir)), self.swap(
+                    build, 'COMPILED_JS_DIR', MOCK_COMPILED_JS_DIR):
+            build.test_compiled_js_dir_is_valid()
+
+    def test_compiled_js_dir_is_deleted_before_compilation(self):
+        """Test that compiled_js_dir is deleted before a fresh
+        compilation.
+        """
+        def mock_check_call(unused_cmd):
+            pass
+        # pylint: disable=unused-argument
```

Comment:
Just prefix the unused args with unused_, then you don't need this pylint pragma.

(I thought I mentioned this previously.)

Reply:
Yes, I used unused_ with cmd but `shell` and `stdout` are passed as named args, so changing the name in mock results in failure of tests.